### PR TITLE
changed: Delete stream details for all childs when movie/musicvideo/e…

### PR DESCRIPTION
**Pull request by arnova**
This is an improved for #16342. It now also removes streamdetails for episodes of tvshows. It also moves the actual stream details removal to the videodatabase delete-functions where they belong.

Note that for a full fix, imo we should start checking the hash/date of files in the thumbloader/extractor. I *may* look at this at some point.

### CODE REVIEW

**Comment by KarellenX** [comment]
Hi @arnova 

Could you please kick off a test build for Win64. 

Thanks :)

**Comment by arnova** [comment]
> Hi @arnova
> 
> Could you please kick off a test build for Win64.
> 
> I also have an Issue report here that lists other problems... #18702
> 
> Thanks :)

This won't fix issues with ISO/VOB/DVD/BR, I guess. Those can only be fixed by, like I suggested, have the thumbloader/extractor check the item's date/hash or be reimplementing the thumb/flag extraction logic in VideoPlayer (which is not my turf nor do I think that that's the way to go).

**Comment by KarellenX** [comment]
Ok. Is there anything you want me to do here? Any testing on this?

**Comment by DaveTBlake** [comment]
Good to see you back @arnova and yes if this solves some of the issues that #16342 introduced then we will have it for RC1.

**Comment by KarellenX** [comment]
Thanks @DaveTBlake 

@arnova 
I read this PR as containing two changes...
1. A fix to allow updating Stream Details over the entire TV Show, rather than each individual episode as I detailed in my Issue report

2. ...
>It also moves the actual stream details removal to the videodatabase delete-functions where they belong.

 I can Delete a movie, episode, tv show from the library and this also deletes the streamdetails. So now I have a choice to either Refresh or Delete to update streamdeatils



Is this correct? This is what I will be testing a bit later today.

**Review by mohdreza1358 — APPROVED** [review]

**Comment by arnova** [comment]
> Thanks @DaveTBlake
> 
> @arnova
> I read this PR as containing two changes...
> 
> 1. A fix to allow updating Stream Details over the entire TV Show, rather than each individual episode as I detailed in my Issue report
> 2. ...
> 
> > It also moves the actual stream details removal to the videodatabase delete-functions where they belong.
> 
> I can Delete a movie, episode, tv show from the library and this also deletes the streamdetails. So now I have a choice to either Refresh or Delete to update streamdeatils
> 
> Is this correct? This is what I will be testing a bit later today.

Correct :-) FYI: The latter change effectively only moves code it doesn't change or add functionality.

**Comment by KarellenX** [comment]
@arnova 

I noticed that this build is using old versions of all the databases. Should it be this way?

**Movies**
I added a movie and scraped it into the library using mkv files. I then deleted the movie, replaced the movie file with a different quality video but named the same, and rescraped it. The new streamdetails were read and displayed.

I then changed the movie file back to the original movie file and Refreshed the movie. The new streamdetails were read and displayed correctly.


**TV Shows**
I scraped a TV Show with 4 *.mkv episodes. The streamdetails were correctly displayed.

I swapped in different video files and Refreshed the TV show including the episodes. All episode streamdetails were correctly updated.

I deleted the TV Show from the library (did not run a Clean Library). Swapped back the original video files and rescraped the show. The correct streamdetails were displayed.

It is good to see that streamdetails are being cleared when an entry is deleted from the library as this was never the case up until now.

Thanks @arnova It all seems to be working as expected

**Review by KarellenX — APPROVED** [review]

**Comment by arnova** [comment]
> @arnova
> 
> I noticed that this build is using old versions of all the databases. Should it be this way?
> 

Yes could be using old db: I didn't bother doing git pull upstream master when branching for this PR. Thanks for testing, great that it's working as intended.

**Comment by arnova** [comment]
@KarellenX (and others): FYI, the plan is to add a hash field to the video files table so we can keep track of modified video files. This will allow us to cleanup any out-of-date extracted details/thumbs upon database cleanup. Obviously this will be for post-Matrix, but this should provide a proper fix for all the remaining issues.

File: xbmc/video/VideoDatabase.cpp [hunk 1]
```diff
@@ -3515,7 +3517,15 @@ void CVideoDatabase::DeleteTvShow(int idTvShow, bool bKeepId /* = false */)
...
     GetPathsForTvShow(idTvShow, paths);
 
-    std::string strSQL=PrepareSQL("SELECT episode.idEpisode FROM episode WHERE episode.idShow=%i",idTvShow);
+    std::string strSQL=PrepareSQL("SELECT episode.idFile FROM episode WHERE episode.idShow=%i",idTvShow);
+    m_pDS2->query(strSQL);
+    while (!m_pDS2->eof())
+    {
+      DeleteStreamDetails(m_pDS2->fv(0).get_asInt());
+      m_pDS2->next();
+    }
```
Comment (Montellese) [inline xbmc/video/VideoDatabase.cpp:3526]: Sorry for being late to the discussion. Can you elaborate why this is necessary here? A few lines below we already call `DeleteEpisode()` which (with your other changes) in turn calls `DeleteStreamDetails()` based on the linked file. So why do we have to do this manually here as well?

Comment (arnova) [reply xbmc/video/VideoDatabase.cpp:3526]: Nice catch! Yes, you're right. I'll correct it.

**Pull request merged**